### ISSUE TITLE
Localized versions

### ISSUE THREAD

**Issue by GoogleCodeExporter** [classilla/tenfourfox#42]
```
* I read everything above and have demonstrated this bug only occurs on
10.4Fx by testing against this official version of Firefox 4 (not
applicable for startup failure) - specify:

* This is a startup crash or failure to start (Y/N): N

* What steps are necessary to reproduce the bug? These must be reasonably 
reliable.

Start tenfourfox on a non english system

* Describe your processor, computer, operating system and any special
things about your environment.

IMac Power PC G4 1,25Ghz

No localization is included into the package. I Try to replace en-en.jar by the 
official fr.jar in chrome directory but didn't work. I manage to change search 
engines, and dictionaries... 

How could we localize tenfourfox ? 



```

Original issue reported on code.google.com by `narcoti...@gmail.com` on 22 Mar 2011 at 10:58
* Blocked on: #61

**Comment by chris-chtrusch** [comment]
[#180 made by kn****n]

The grammar of "insticksmodul" is correct. Insticksmodul and tillägg has the same meaning when used for the english word "plugin", so either one works gramatically.

If the official Swedish FFX translation uses "insticksmodul", use that one.

**Comment by chris-chtrusch** [comment]
Test installer for German localization.

**Summary of changes for 45:**
- TenFourFox.dtd and TenFourFox.properties contain strings of features to be added to TFF at a later stage. 
- string "TenFourFox does not support plugins for this content." has been moved to chrome/(language)/locale/(language)/pluginproblem/pluginproblem.dtd

**Bug List:**
- Help Menu is still missing blue search bar on 10.5
- Help Section in Hamburger Menu is empty (10.4 & 10.5)


We, once again, need volunteers for these new strings to be translated to Finnish, French, Italian, Polish, Russian, Spanish, Swedish:

==new strings for 45==
User Agent
Set user agent to current version of Firefox
Select user agent string to use:
TenFourFox 45 (default)
Adblock
Enable basic adblock
PDF Viewing
Use built-in PDF viewer (slower but safer)
================

Please keep these translations ar short as possible.

**Work schedule/assignment for localizations TFF 45**

German: me
French: grafiko… ?
Spanish: me, but need translator for new strings, see above
Italian: mcl… ?
Russian: me
Polish: aquarius… ?
Finnish: me, but need translator for new strings, see above
Japanese: ???
[TFF-45.x-Sprachpaket-Deutsch-test1.app.zip](https://github.com/classilla/tenfourfox/files/404950/TFF-45.x-Sprachpaket-Deutsch-test1.app.zip)

**Comment by chris-chtrusch** [comment]
Test locale installer German for TFF 45

[Uploading TFF-45.x-Sprachpaket-Deutsch-test1.app.zip…]()

**Comment by grafikoloco** [comment]
Hi Chris,
Locale installer French TTF 45.

[TFF-45.x-packlangage-Français.zip](https://github.com/classilla/tenfourfox/files/405235/TFF-45.x-packlangage-Francais.zip)

**Comment by chris-chtrusch** [comment]
Perfect, thank you. 
German installer (final):
[TFF-45.x-Sprachpaket-Deutsch.app.zip](https://github.com/classilla/tenfourfox/files/405331/TFF-45.x-Sprachpaket-Deutsch.app.zip)

**Comment by classilla** [comment]
This is getting a little long. Do we want another issue specifically for 45?

**Comment by chris-chtrusch** [comment]
You can split it off from 42#issuecomment-238014693 if necessary.

**Comment by chris-chtrusch** [comment]
Spanish installer for 45
[TFF-45.x-Langpack-Spanish.app.zip](https://github.com/classilla/tenfourfox/files/428221/TFF-45.x-Langpack-Spanish.app.zip)

**Comment by mclcmm** [comment]
Italian installer for 45
[TFF-45.x-Langpack-Italiano.zip](https://github.com/classilla/tenfourfox/files/428599/TFF-45.x-Langpack-Italiano.zip)
At the moment I'm away from my powerpc macs, so I can't test the langpack myself. Could you please test it for me and check everything is in order? Thanks!

**Comment by chris-chtrusch** [comment]
I'm sorry, it doesn't work at all. I can't find the error, though. Looks like a different version of Firefox was used, not Intel-Firefox 45.x ESR.

**Comment by mclcmm** [comment]
I used Firefox 45.3.0 ESR. I'll try to get my hands on a powerpc mac and debug the installer, but this might take a while.

**Comment by chris-chtrusch** [comment]
Russian installer for 45.

**Comment by chris-chtrusch** [comment]
Russian installer for 45.
[TFF-45.x-Langpack-Russian.zip](https://github.com/classilla/tenfourfox/files/440539/TFF-45.x-Langpack-Russian.zip)

**Comment by chris-chtrusch** [comment]
Swedish installer for 45.
[TFF-45.x-Langpack-Swedish.zip](https://github.com/classilla/tenfourfox/files/440720/TFF-45.x-Langpack-Swedish.zip)

**Comment by chris-chtrusch** [comment]
Mlc: I think I found the bug in your installer: You have invalid special characters in at least two of your modified string files (baseMenuOverlay and about dialog), probably all of them. Possibly do to opening the Intel files in MacOS Roman and saving them in UTF 8 or the other way around. If you have the time to fix it, please try to get the installer ready before September 10. If you can't make it, I can do the installer.

**Comment by classilla** [comment]
Chris, I've been thinking about MediaSource and maybe we'll just expose that as a user-visible setting (to be implemented). Can we add, to the TenFourFox.dtd and .properties file for these other translations,

`<!ENTITY TFFmseMode.title       "MediaSource">`
`<!ENTITY TFFmseMode.prompt      "Enable MSE/media quality options (if available)">`
`TFFmseMode.title.p      = MediaSource`
`TFFmseMode.prompt.p     = Enable MSE/media quality options (if available)`

**Comment by chris-chtrusch** [comment]
That's a very good idea, but these strings would probably stay English for most languages until we find volunteer translators here. (We could replace the installers once they're ready.) I've had the other new strings translated as a little deal between me and some collegues in the translation business in order to get the installers finished on time, but I won't be able to repeat this any time soon.

**Comment by mclcmm** [comment]
Revised version of the Italian langpack installer:
[TFF-45.x-Langpack-Italiano.zip](https://github.com/classilla/tenfourfox/files/440923/TFF-45.x-Langpack-Italiano.zip)

It seems to work on this iBook, but please once again check everything is in order. Thanks!
I left out the new strings about MediaSource, but I'll attach a translation in case you want to add them at a later date:

<!ENTITY TFFmseMode.title "MediaSource">
<!ENTITY TFFmseMode.prompt "Abilita le opzioni sulla qualità MSE/media (lento, se disponibile)">
TFFmseMode.title.p = MediaSource
TFFmseMode.prompt.p = Abilita le opzioni sulla qualità MSE/media (lento, se disponibile)

**Comment by chris-chtrusch** [comment]
This looks very good now, thank you!

**Comment by chris-chtrusch** [comment]
I'm going to make Finnish and possibly Polish installers on the next weekend, then we'll have the same set of localizations available as we had for 38. If Cameron decides to put the MediaSource strings in the release version, I'll re-post all installers with either English, Google translated or (preferably) human translated versions of this string. If anyone can contribute a translation to these strings, please post it here.

**Comment by grafikoloco** [comment]
Hello Chris,
I added the translation MediaSource.
[TFF-45.x-Packlangage-Français.zip](https://github.com/classilla/tenfourfox/files/441160/TFF-45.x-Packlangage-Francais.zip)

**Comment by chris-chtrusch** [comment]
Polish and Finnish installers. 
[TFF-45.x-Langpack-Finnish.zip](https://github.com/classilla/tenfourfox/files/453401/TFF-45.x-Langpack-Finnish.zip)
[TFF-45.x-Polski_pakiet_jezykowy.zip](https://github.com/classilla/tenfourfox/files/453400/TFF-45.x-Polski_pakiet_jezykowy.zip)

**Comment by chris-chtrusch** [comment]
Please contribute translations for this string:

_Enable MSE/media quality options (slower, if available)_

…in Finnish, Polish, Russian, Spanish and Swedish if you are a native speaker of these languages. Google Translate doesn't seem to have enough technical understanding for this, it seems, so this string will stay English for release if we don't get human translations by Sept. 10 (French, Italian and German are complete by now).

NB: Most users have no idea what MSE is, and the term "Media" is quite generic. Also, Mediasource Extensions can hardly be translated in a meaningful way; most international bloggers/tech writers just use the English term. Therefore the translation should give a hint that users will be able to choose different video sizes if they enable this.

I will upload the final installers in one post Sept. 10., so they'll be ready for the 45.4 release on Sept. 13.

**Comment by classilla** [comment]
Yes, please upload today; the test release will come out this afternoon and I need to start work on the new front page.

**Comment by chris-chtrusch** [comment]
Installers for 45 release:
[TFF-45.x-Langpack-Swedish.zip](https://github.com/classilla/tenfourfox/files/465545/TFF-45.x-Langpack-Swedish.zip)
[TFF-45.x-Langpack-Finnish.zip](https://github.com/classilla/tenfourfox/files/465547/TFF-45.x-Langpack-Finnish.zip)
[TFF-45.x-Packlangage-Français.zip](https://github.com/classilla/tenfourfox/files/465546/TFF-45.x-Packlangage-Francais.zip)
[TFF-45.x-Sprachpaket-Deutsch.zip](https://github.com/classilla/tenfourfox/files/465548/TFF-45.x-Sprachpaket-Deutsch.zip)
[TFF-45.x-Langpack-Italiano.zip](https://github.com/classilla/tenfourfox/files/465549/TFF-45.x-Langpack-Italiano.zip)
[TFF-45.x-Polski_pakiet_jezykowy.zip](https://github.com/classilla/tenfourfox/files/465550/TFF-45.x-Polski_pakiet_jezykowy.zip)
[TFF-45.x-Langpack-Russian.zip](https://github.com/classilla/tenfourfox/files/465552/TFF-45.x-Langpack-Russian.zip)
[TFF-45.x-Langpack-Spanish.zip](https://github.com/classilla/tenfourfox/files/465551/TFF-45.x-Langpack-Spanish.zip)

Note: I corrected an "error" in the French installer app, which still said TFF 38. French, German and Italian have the MSE string localized. The other installers have it in English and will get localized versions as we find volunteer translators over time (come on, it's only one line…!), preferably before the MSE toggle is implemented :-) In the worst case, the installers will continue to function but will show this one preference in English. Right now, the problem isn't even visible to the user, so I think it's better to release them the way they are instead of offering no langpacks at all.

**Comment by classilla** [comment]
That works. I'll give a couple hours grace period for people to catch up and then upload to SourceForge either tonight or tomorrow Pacific time. The MSE toggle (and the PDF toggle) are scheduled for 45.5. More on that later. Once the langpacks are posted, I will close this issue and set up a new one for new strings for feature parity, as those occur.

**Comment by chris-chtrusch** [comment]
Good. Strings for feature parity are easier because they are already translated in the official Firefox. It would be good to know your plans for version numbering from now on to adjust the version checking routine in the installer app. I.e. will we move from 45.x to 46.x when new features (for parity) are added?

**Comment by classilla** [comment]
Haven't decided yet. We can discuss that in the new thread (#328).

**Comment by bovh** [comment]
"Enable MSE/media quality options (slower, if available)" --> "Salli MSE/medialaatuoptiot (hitaampi, mikäli tarjolla)". Literal but Finnish.

**Comment by bovh** [comment]
"Enable MSE/media quality options (slower, if available)" --> "Ota käyttöön MSE/medialaatuoptiot (hitaampi, mikäli tarjolla)". This might be better. (Enable ~ ota käyttöön. Allow ~ salli.)

**Comment by chris-chtrusch** [comment]
Thank you. I guess we could keep this issue open for completion of the MSE string translation and use 328 for everything after that. I'll make a post in which completed installers are collected, to be uploaded to Sourceforge before 45.5.

**Comment by chris-chtrusch** [comment]
**Installers completed with MSE translated string.**
Still missing: Spanish, Polish, Russin, Swedish.

[TFF-45.x-Langpack-Finnish.zip](https://github.com/classilla/tenfourfox/files/470579/TFF-45.x-Langpack-Finnish.zip)

**Comment by classilla** [comment]
Sounds good to me.

**Comment by bovh** [comment]
That latest Finnish langpack did not make it to sourceforge.

**Comment by classilla** [comment]
> That latest Finnish langpack did not make it to sourceforge.

Just to confirm: the langpack in this ticket does work?

**Comment by chris-chtrusch** [comment]
The langpack from Sep 14 works and is the version that was intended for release.

**Comment by classilla** [comment]
> The langpack from Sep 14 works and is the version that was intended for release.

Okay, then the screwup was probably mine. I'll reupload it when I get a chance.

**Comment by chris-chtrusch** [comment]
Langpacks need a fix for Open/Save/Print dialogs, which are incompletely localized in the portions that are controlled by the OS. 

This is easily done by adding an lproj folder with an additional InfoPlist.strings file. [Graficoloco or Victorien hinted me to this a long time ago but I simply forgot it and never saw the problem because all my Mac OS X installations are English anyway.]

I'm going to post updated installers here shortly for all languages. We could exchange the affected installers right away or just slip the fix in when the next custom string update is needed.

**Comment by chris-chtrusch** [comment]
Updated langpack Installers with fix for Print/Open/Save dialog localizations. Also updated the copyright date in the About dialog to 'Copyright © 2010-2018'

[TFF-45.x-Langpack-Finnish.app.zip](https://github.com/classilla/tenfourfox/files/1611815/TFF-45.x-Langpack-Finnish.app.zip)
[TFF-45.x-Packlangage-Français.app.zip](https://github.com/classilla/tenfourfox/files/1611816/TFF-45.x-Packlangage-Francais.app.zip)
[TFF-45.x-Sprachpaket-Deutsch.app.zip](https://github.com/classilla/tenfourfox/files/1611817/TFF-45.x-Sprachpaket-Deutsch.app.zip)
[TFF-45.x-Langpack-Italiano.app.zip](https://github.com/classilla/tenfourfox/files/1611818/TFF-45.x-Langpack-Italiano.app.zip)
[TFF-45.x-Polski_pakiet_jezykowy.app.zip](https://github.com/classilla/tenfourfox/files/1611819/TFF-45.x-Polski_pakiet_jezykowy.app.zip)
[TFF-45.x-Langpack-Russian.app.zip](https://github.com/classilla/tenfourfox/files/1611820/TFF-45.x-Langpack-Russian.app.zip)
[TFF-45.x-Langpack-Spanish.app.zip](https://github.com/classilla/tenfourfox/files/1611821/TFF-45.x-Langpack-Spanish.app.zip)
[TFF-45.x-Langpack-Swedish.app.zip](https://github.com/classilla/tenfourfox/files/1611822/TFF-45.x-Langpack-Swedish.app.zip)

**Comment by classilla** [comment]
Thanks, I'll update SourceForge when I get back from vacation (running out the door, will have limited Internet for a few days). Since this is getting unwieldy, let's put further updates in #328.

**Comment by classilla** [comment]
Actually, since we're regenerating them: could you also add the code to _also_ accept FPRx as a version?

**Comment by classilla** [comment]
Test against FPR4.1. It won't get more after that. It should work with FPRx and FPRx.y.

**Comment by grafikoloco** [comment]
Hello, Chris,

Thank you for this update.

And happy new year 2018.

kind regards
grafiko

--
Cordialement
grafikoloco

Tel : 0596 62 25 05
grafikoloco@free.fr
http://grafikoloco.free.fr

**Comment by classilla** [comment]
Langpacks now live on site. Please follow up in #328 for additional localization requests.

**Comment by chris-chtrusch** [comment]
Cameron, the langpacks in issue 328 are more recent (with updated version check routine).

**Comment by classilla** [comment]
Dammit. I'll redo these today.
